Data/TLS/PThread.hs

```diff
@@ -15,6 +15,7 @@
 module Data.TLS.PThread
     ( TLS
     , mkTLS
+    , mkTLSWithDestructor
```

Comment:
Is there a GHC equivalent of this functionality? I only ask since currently `thread-local-storage` claims to "provide exactly the same interface with different implementations", but this claim would no longer be true if we have `mkTLSWithDestructor` in `Data.TLS.PThread` but not `Data.TLS.GHC`.

I can see two ways around this:

1. Implement `mkTLSWithDestructor` in `Data.TLS.GHC`.
2. Only export `mkTLSWithDestructor` from `Data.TLS.PThread.Internal`.

If we pick the second option, make sure to add an explicit type signature for `mkTLSWithDestructor`, since it won't be specified in `TLS_Sig.hs`.